packages/@aws-cdk/aws-cognito/.eslintrc.js

```diff
@@ -1,3 +1,24 @@
 const baseConfig = require('cdk-build-tools/config/eslintrc');
 baseConfig.parserOptions.project = __dirname + '/tsconfig.json';
-module.exports = baseConfig;
+
+const config = {
+  ...baseConfig,
+  plugins: [
+      ...baseConfig.plugins,
+      'node',
+  ],
+  rules: {
+      ...baseConfig.rules,
+      'node/no-restricted-import': [
+          'error',
+          [
+              {
+                  name: ['punycode', '!punycode/'],
+                  message: `Package 'punycode' has to be imported with trailing slash, see warning in https://github.com/bestiejs/punycode.js#installation`,
+              },
+          ],
+      ],
```

Comment:
Probably safe to move this rule to the global eslintrc config - `cdk-build-tools/config/eslintrc` - and apply this across the entire repo.